### PR TITLE
Generate upload instructions in the PR for secrets and bytes

### PR DESCRIPTION
**Pull request by mandopaloooza**
## Description
Added upload instructions to entity that needs to be loaded from an external source. 

When computing the commit message for each entity, we check if `appgate_metadata["url_file_path"] is set. If it is set, print instructions specifying what path to upload the file/secret contents of the external storage. 

I'm using `url_file_path()` which is also used by the loader to figure out what URL to fetch the contents. 

Screenshot 2023-02-21 at 4 19 47 PM

## Checklist
- [x] Bump `.version` and `.appVersion` in [k8s/crd/Chart.yaml](../k8s/crd/Chart.yaml)
- [x] Bump `.version` and `.appVersion` in [k8s/operator/Chart.yaml](../k8s/operator/Chart.yaml)

### CODE REVIEW

**Review by thomascellerier — APPROVED** [review]
Looks good.
What do you think of making it a checkbox even? So it acts as a checklist before merging the pull request.

**Comment by mandopaloooza** [comment]
> What do you think of making it a checkbox even? So it acts as a checklist before merging the pull request.

Good idea!

**Comment by mandopaloooza** [comment]
Screenshot 2023-02-22 at 9 54 55 AM

@thomascellerier What do you think of this format?

**Comment by thomascellerier** [comment]
Looks good!

**Comment by thomascellerier** [comment]
Does it work in gitlab too?

**Comment by mandopaloooza** [comment]
@thomascellerier  Verified it works on GitLab as well

GitLab: 
Screenshot 2023-02-22 at 4 00 15 PM

GitHub:
Screenshot 2023-02-22 at 4 00 44 PM

**Comment by mandopaloooza** [comment]
@lepiolet The flow to print the upload instructions in the PR is as follows:
1. When loading a byte/secret entity, set `appgate_metadata["urlFilePath"]` to save the URL. For bytes, this is done in `set_appgate_file_metadata()` and for secrets, it is done in `set_appgate_password_metadata()`
2. When getting the git commit message, we check if the entity's appgate metadata contains a non-null `url_file_path`. If this metadata is available, add a checkmark box with instructions on where to upload

File: appgate/files.py [hunk 1]
```diff
@@ -71,19 +76,32 @@ def url_file_path(
...
-    if (v := value.get(field_name)) is not None:
+    v = value.get(field_name)
+    if v is not None and isinstance(v, str):
         return normalize_url_file_path(v)
     for field in target_fields:
         if v := value.get(field):
             return v
     if value.get("name"):
         return f"{normalize_url_file_path(value['name'])}/{field_name}"
-    elif value.get("id"):
```
Comment (lepiolet) [inline appgate/files.py:81]: :D

File: appgate/files.py [hunk 2]
```diff
@@ -71,19 +76,32 @@ def url_file_path(
...
+    if value.get("id"):
         return f"{value['id']}/{field_name}"
-    else:
-        raise AppgateFileException(
-            f"Unable to generate url to get fetch file for field {field_name} for entity {entity_name}"
-        )
+
+    # TODO: Problematic entities. Below are hardcoded workarounds.
+    if entity_name == "Appliance_Ntp_Servers":
```
Comment (lepiolet) [inline appgate/files.py:91]: Good enough for now but we should implement a generic `x-field` to catch this kind of problems

File: appgate/files.py [hunk 3]
```diff
@@ -215,6 +247,15 @@ def name_or_id(values: dict[str, Any]) -> str:
                 ),
                 field=self.name,
                 load_external=should_load_file(self.operator_mode),
-            )
+            ),
+            CustomEntityLoader(
```
Comment (lepiolet) [inline appgate/files.py:251]: mmmm can not the new field be added into the appgate metadata directly? is really needed a new custom loader? Think that if we have it in the appgate metadata from the beginning we could simplify the FileAttribLoader

File: appgate/secrets.py [hunk 4]
```diff
@@ -282,10 +298,26 @@ def set_appgate_password_metadata(orig_values, entity: Entity_T) -> Entity_T:
                 field=self.name,
                 load_external=should_load_secret(self.operator_mode),
             ),
-            CustomEntityLoader(loader=set_appgate_password_metadata),
```
Comment (lepiolet) [inline appgate/secrets.py:285]: Same as before

File: appgate/syncer/git.py [hunk 5]
```diff
@@ -699,18 +689,27 @@ async def _create(self, e: Entity_T, register_commit: bool = True) -> EntityClie
         p.mkdir(exist_ok=True)
         file = git_dump(e, self.api_spec, p, self.resolution_conflicts)
         if register_commit:
-            self.commits.append((file, "ADD"))
+            self.commits.append(
+                (p, GitCommitState(entity=e, path=file, operation="ADD"))
```
Comment (lepiolet) [inline appgate/syncer/git.py:693]: we were returning `file` now we return `p`. Is that on purpose?

**Review by lepiolet — APPROVED** [review]
Looks good, just some questions

File: appgate/syncer/git.py (hunk 5 above)
Comment (mandopaloooza) [reply appgate/syncer/git.py:693]: Unintentional. Reverting back to `file`

**Pull request merged**